File: src/main/java/com/architecture_map/belarus/controller/ArchitectController.java
package com.architecture_map.belarus.controller;

import com.architecture_map.belarus.dto.ArchitectDto;
import com.architecture_map.belarus.entity.Architect;
import com.architecture_map.belarus.exception.NotFoundException;
import com.architecture_map.belarus.service.ArchitectService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/architects")
@RequiredArgsConstructor
@CrossOrigin(origins = {"http://localhost:7200", "http://localhost:4200"})
public class ArchitectController {

    private final ArchitectService architectService;

    @PostMapping
    public ResponseEntity<String> create(@RequestBody ArchitectDto architectDto) {
        Architect savedArchitect = architectService.create(architectDto);

        HttpHeaders headers = new HttpHeaders();
        headers.add("Location", savedArchitect.getId().toString());

        return new ResponseEntity<>(headers, HttpStatus.CREATED);
    }

    @GetMapping("/{id}")
    public Architect findById(@PathVariable Integer id) {
        return architectService.findById(id).orElseThrow(NotFoundException::new);
    }

    @GetMapping("/")
    public List<Architect> findAll(){
        return architectService.findAll();
    }
}

File: src/main/kotlin/by/architecturemap/belarus/controller/ArchitectController.kt
package by.architecturemap.belarus.controller

import by.architecturemap.belarus.entity.Architect
import by.architecturemap.belarus.service.ArchitectService
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.CrossOrigin
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/architects")
@CrossOrigin(origins = ["http://localhost:7200", "http://localhost:4200", "*"])
class ArchitectController(
    private val architectService: ArchitectService
) {

    @PostMapping
    fun create(@Valid @RequestBody architect: Architect): ResponseEntity<Architect> =
        ResponseEntity(architectService.create(architect), HttpStatus.CREATED)

    @GetMapping("/{id}")
    fun find(@PathVariable id: Int): Architect = architectService.find(id)

    @GetMapping("/")
    fun findAll(): List<Architect> = architectService.findAll()
}
